server/src/main/java/org/opensearch/cluster/routing/remote/NoopRemoteRoutingTableService.java

```diff
@@ -39,15 +38,14 @@ public DiffableUtils.MapDiff<String, IndexRoutingTable, Map<String, IndexRouting
     }
 
     @Override
-    public CheckedRunnable<IOException> getAsyncIndexRoutingWriteAction(
+    public void getAsyncIndexRoutingWriteAction(
         String clusterUUID,
         long term,
         long version,
         IndexRoutingTable indexRouting,
         LatchedActionListener<ClusterMetadataManifest.UploadedMetadata> latchedActionListener
```

Comment:
who triggers the latched listener in case of no-op?

Reply:
As same as before, in case of No-Op listener the listener is not called back. We will not be calling this method if RemoteRoutingTableService is instance of NoopRemoteRoutingWriteAction, this method will not be invoked.